src/utils_db_query.c

```diff
@@ -1011,7 +1012,13 @@ int udb_query_prepare_result (udb_query_t const *q, /* {{{ */
   if ((q == NULL) || (prep_area == NULL))
     return (-EINVAL);
 
-  udb_query_finish_result (q, prep_area);
```

Comment:
Uh, I missed that, thanks. Anyway, in that case I actually disagree strongly with the proposed change. udb_query_prepare_result() overwrites the content in the preparation area and, hence, it's right to cleanup the old content before. It's totally fine for the caller not to care about this and neither a memory leak (if compiled without debugging) or a program crash would be acceptable.